lib/HTFeed/Stage/ImageRemediate.pm

```diff
@@ -369,7 +370,107 @@ sub _remediate_tiff {
       && $self->repair_tiff_exiftool( $infile, $outfile, $self->{newFields} );
 
     return $ret;
+}
+
+# Extract metadata from jpg/png and write to the corresponding tiff.
+sub copy_metadata {
+    my $self    = shift;
+    my $ext     = shift;
+    my $infile  = shift;
+    my $outfile = shift;
+
+    $self->{oldFields} = $self->get_exiftool_fields($infile);
+    $self->{newFields} = {};
+
+    # Delegate to the method that knows how to extract from a ".$ext"
+    if ($ext eq ".jpg") {
+	$self->{newFields} = extract_jpg_metadata($self->{oldFields});
+    } elsif ($ext eq ".png") {
+	$self->{newFields} = extract_png_metadata($self->{oldFields});
+    } else {
+	croak "copy_metadata knows not extension: $ext";
+	return;
+    }
+
+    # Write extracted metadata to outfile.
+    my $exifTool = new Image::ExifTool;
+    while (my ($field, $val) = each(%{$self->{newFields}})) {
+	my ($success, $errStr) = $exifTool->SetNewValue($field, $val);
+	if (defined $errStr) {
+	    croak("Error setting new tag $field => $val: $errStr\n");
+	    return 0;
+	}
+    }
+    my $exif_write_status = $exifTool->WriteInfo($outfile);
+    unless ($exif_write_status == 1) {
+	get_logger()->trace("Failed EXIF write to $outfile");
+    }
+}
+
+# Build and return a hash of jpg metadata.
+sub extract_jpg_metadata {
+    my $olf = shift; # ref to $self->{oldFields}, a hash of exiftool data
+    my $h = {
+	'IFD0:ModifyDate'         => $olf->{'System:FileModifyDate'},
```

Comment:
As below I don't think we need to copy `ImageHeight`, `ImageWidth`, `BitsPerSample`. I think `dc:source`, `IFD0:ModifyDate`, `tiff:DateTime` probably aren't necessary because we normalize those later and use whatever information we can find (capture date from the METS or fallback to filesystem timestamp if need be, I think).

Reply:
OK, not getting any of them.